### PR TITLE
Fixed #858, FromConfigRegistryAuthHandlerTest: Use assertThrows instead of deprecated ExpectedException.none() @Rule

### LINKED ISSUE #858: FromConfigRegistryAuthHandlerTest: Use `assertThrows` instead of deprecated `ExpectedException.none()` `@Rule`
As pointed out in [JUnit Javadocs](https://junit.org/junit4/javadoc/latest/org/junit/rules/ExpectedException.html#none()), `ExpectedException.none()` rule is deprecated. We should use `Assert.assertThrows` instead

https://github.com/eclipse/jkube/blob/5a1828c0ece30567fdcf89ce7668bb8c2003211d/jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java#L42-L43

We need to refactor `testFromPluginConfigurationFailed()`. Here is what you need to do:
- [x] Remove `ExpectedException.none()` `@Rule` and use `assertThrows`
- [x] Use AssertJ's `assertThat` instead of Hamcrest assertion for asserting exception message `org.hamcrest.CoreMatchers.containsString`

### PR DESCRIPTION
**Pull request by soumyadeeptadas**
# Fixed [ issue #858](https://github.com/eclipse/jkube/issues/858)

## Tasks Completed:
- 1.  Remove ExpectedException.none() @Rule and use assertThrows    : Completed
- 2. Use AssertJ's assertThat instead of Hamcrest assertion for asserting exception message org.hamcrest.CoreMatchers.containsString     : Completed

## Overall changes
- imports for assertJ assertThat() and Junit assertThrows()
- commented out deprecated usage rules for expectedException
- refactored testFromPluginConfigurationFailed() function with assertThat and assertThrows

### CODE REVIEW

File: jkube-kit/build/api/pom.xml [hunk 1]
```diff
@@ -31,6 +31,14 @@
 
   <dependencies>
 
+      <dependency>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter-api</artifactId>
+        <version>5.7.0</version>
+        <scope>test</scope>
+    </dependency>
```
Comment (rohanKanojia) [inline jkube-kit/build/api/pom.xml:39]: why is this needed?

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java [hunk 2]
```diff
@@ -39,8 +44,8 @@
     @Mocked
     private KitLogger log;
 
-    @Rule
-    public ExpectedException expectedException = ExpectedException.none();
+    /*@Rule
+    public ExpectedException expectedException = ExpectedException.none();*/      //removed expectedException @Rule
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:48]: Instead of commenting them out, please remove it.

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java [hunk 3]
```diff
@@ -82,8 +87,15 @@ public void testFromPluginConfigurationFailed() {
         FromConfigRegistryAuthHandler handler = new FromConfigRegistryAuthHandler(
             RegistryAuthConfig.builder().putDefaultConfig(RegistryAuth.USERNAME, "admin").build(), log);
 
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage(containsString("password"));
+        //expectedException.expect(IllegalArgumentException.class);
+        IllegalArgumentException exception= assertThrows(IllegalArgumentException.class,() -> { throw new IllegalArgumentException("throw illegal arg exception"); });      
+        //expectedexception implementation substituted with assertThrows #task1
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:92]: please remove commented out lines

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java [hunk 4]
```diff
@@ -30,6 +30,11 @@
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.junit.Assert.assertEquals;
 
+
+
+import static org.assertj.core.api.Assertions.assertThat;  /*assertj assertThat import*/
+import static org.junit.jupiter.api.Assertions.assertThrows;   /*assertThrows import*/
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:36]:
```suggestion
import static org.junit.Assert.assertThrows;
```

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java [hunk 5]
```diff
@@ -82,8 +87,15 @@ public void testFromPluginConfigurationFailed() {
         FromConfigRegistryAuthHandler handler = new FromConfigRegistryAuthHandler(
             RegistryAuthConfig.builder().putDefaultConfig(RegistryAuth.USERNAME, "admin").build(), log);
 
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage(containsString("password"));
+        //expectedException.expect(IllegalArgumentException.class);
+        IllegalArgumentException exception= assertThrows(IllegalArgumentException.class,() -> { throw new IllegalArgumentException("throw illegal arg exception"); });
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:91]:
```suggestion
        IllegalArgumentException exception= assertThrows(IllegalArgumentException.class,() -> handler.create(RegistryAuthConfig.Kind.PUSH, null, null, s -> s); });      
```

File: jkube-kit/build/api/pom.xml (hunk 1 above)
Comment (soumyadeeptadas) [reply jkube-kit/build/api/pom.xml:39]: The import for assertThrows from "junit-jpuiter-api" was requiring specification of that dependency in pom.xml.

Since you have already suggested the changes, ill use the Assert.assertThrows import

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java (hunk 3 above)
Comment (soumyadeeptadas) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java]: removed

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java (hunk 2 above)
Comment (soumyadeeptadas) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java]: removed

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java (hunk 5 above)
Comment (soumyadeeptadas) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java]: updated

**Comment by rohanKanojia** [comment]
Please make sure you have created an eclipse account and have a signed off by footer in your commit message. https://github.com/eclipse/jkube/pull/856#issuecomment-900181000

**Comment by rohanKanojia** [comment]
Did you verify your changes are compiling okay by doing `mvn clean install` ?

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java (hunk 4 above)
Comment (soumyadeeptadas) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java]: modified

**Comment by soumyadeeptadas** [comment]
According to [this](https://github.com/eclipse/jkube/pull/859#issuecomment-903662680) , Updated with commit signature (signoff), latest commit shows verified.

**Comment by soumyadeeptadas** [comment]
In reference to [this comment](https://github.com/eclipse/jkube/pull/859#issuecomment-903663236)

The Build is successful, but there are some 15 tests which are failing on running `mvn clean install`, even on the base `original eclipse/jkube repository`. The same status shows for the project after the changes done by me.

My IDE is setup according according to the contributing guidelines.

**Comment by manusa** [comment]
```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   FromConfigRegistryAuthHandlerTest.testFromPluginConfigurationFailed:91 » IllegalArgument
[INFO] 
[ERROR] Tests run: 92, Failures: 0, Errors: 1, Skipped: 0
```

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java [hunk 6]
```diff
@@ -30,6 +30,11 @@
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.junit.Assert.assertEquals;
 
+
+
+import static org.assertj.core.api.Assertions.assertThat;  /*assertj assertThat import*/
+import static org.junit.Assert.assertThrows;   /*assertThrows import*/
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:36]:
```suggestion
import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.Assert.assertThrows;  
```

Comment (rohanKanojia) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:36]: Also, please remove unused imports:
```java
import org.junit.Rule;
import org.junit.rules.ExpectedException;
import static org.hamcrest.CoreMatchers.containsString
```

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java [hunk 7]
```diff
@@ -81,9 +84,10 @@ public void testFromPluginConfigurationPull() {
     public void testFromPluginConfigurationFailed() {
         FromConfigRegistryAuthHandler handler = new FromConfigRegistryAuthHandler(
             RegistryAuthConfig.builder().putDefaultConfig(RegistryAuth.USERNAME, "admin").build(), log);
-
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage(containsString("password"));
+        
+        IllegalArgumentException exception= assertThrows(IllegalArgumentException.class,() -> {handler.create(RegistryAuthConfig.Kind.PUSH, null, null, s -> s); });
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:88]: No need to have curly braces around single lambda statements

```suggestion
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> handler.create(RegistryAuthConfig.Kind.PUSH, null, null, s -> s));            
```

**Review by rohanKanojia — CHANGES_REQUESTED** [review]

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java (hunk 6 above)
Comment (soumyadeeptadas) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java]: updated

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java (hunk 7 above)
Comment (soumyadeeptadas) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java]: updated

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java [hunk 8]
```diff
@@ -81,9 +80,10 @@ public void testFromPluginConfigurationPull() {
     public void testFromPluginConfigurationFailed() {
         FromConfigRegistryAuthHandler handler = new FromConfigRegistryAuthHandler(
             RegistryAuthConfig.builder().putDefaultConfig(RegistryAuth.USERNAME, "admin").build(), log);
-
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage(containsString("password"));
+        
+        IllegalArgumentException exception= assertThrows(IllegalArgumentException.class, () -> handler.create(RegistryAuthConfig.Kind.PUSH, null, null, s -> s));
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/handler/FromConfigRegistryAuthHandlerTest.java:84]: You've put `handler.create()` call in `assertThrows` now but you forgot to remove previous `handler.create` call(on line 87).

This is causing test to fail. Please remove it.

**Comment by rohanKanojia** [comment]
For small changes like these, it's better to keep PR limited to 1 commit. Could you please squash your commits into one using interactive rebase `git rebase -i`?

**Comment by rohanKanojia** [comment]
polite ping, Are you still working on this patch?

**Comment by soumyadeeptadas** [comment]
Im caught up with something else, so im unable to look at it right now
